### PR TITLE
[Hyperdrive] `nodejs_compat` update

### PR DESCRIPTION
**Pull request by Oxyjun**
### Summary

For **Hyperdrive** chapters only:

1. Minimising the number of references to `compatibility_flags` and `compatibility_date` if not directly relevant.
2. Creating a new partial file within `/workers/` which shows the `toml` file which includes the `nodejs_compat` and `2024-09-23` compatibility flag/dates
3. Using partial file within `hyperdrive/configuration/connect-to-postgres`

### Documentation checklist

- [x] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.
- [ ] If a larger change - such as adding a new page- an issue has been opened in relation to any incorrect or out of date information that this PR fixes.
- [ ] Files which have changed name or location have been allocated [redirects](https://developers.cloudflare.com/pages/configuration/redirects/#per-file).

### CODE REVIEW

File: src/content/docs/hyperdrive/tutorials/serverless-timeseries-api-with-timescale/index.mdx [hunk 1]
```diff
@@ -145,8 +145,8 @@ This command outputs your Hyperdrive ID. You can now bind your Hyperdrive config
 ```toml
 name = "timescale-api"
 main = "src/index.ts"
-compatibility_date = "2024-08-21"
-compatibility_flags = [ "nodejs_compat_v2"]
+compatibility_date = "DATE"
```
Comment (thomasgauvin) [inline src/content/docs/hyperdrive/tutorials/serverless-timeseries-api-with-timescale/index.mdx]:
```suggestion
compatibility_date = "2024-09-23"
```

File: src/content/docs/hyperdrive/configuration/connect-to-postgres.mdx [hunk 2]
```diff
@@ -48,7 +47,7 @@ Hyperdrive uses Workers [TCP socket support](/workers/runtime-apis/tcp-sockets/#
...
 | Drizzle                       | [https://orm.drizzle.team/](https://orm.drizzle.team/)                       | `0.26.2`^                |                                                                                                                                                                                                              |
 | Kysely                        | [https://kysely.dev/](https://kysely.dev/)                                   | `0.26.3`^                |                                                                                                                                                                                                              |
 | [rust-postgres](https://github.com/sfackler/rust-postgres) | [https://docs.rs/postgres/latest/postgres/](https://docs.rs/postgres/latest/postgres/) | `v0.19.8` | Use the [`query_typed`](https://docs.rs/postgres/latest/postgres/struct.Client.html#method.query_typed) method for best performance. |
```
Comment (daisyfaithauma) [inline src/content/docs/hyperdrive/configuration/connect-to-postgres.mdx:55]: Not sure if you even can do this but is there a way to view the table without having to scroll?

**Review by daisyfaithauma — APPROVED** [review]

File: src/content/docs/hyperdrive/configuration/connect-to-postgres.mdx (hunk 2 above)
Comment (Oxyjun) [reply src/content/docs/hyperdrive/configuration/connect-to-postgres.mdx:55]: Hmm. I'm not sure what the best practice would be for this (is it manual spacing? Or something else?). Will put in team crit and resolve here for now.

File: src/content/docs/hyperdrive/configuration/connect-to-postgres.mdx [hunk 3]
```diff
@@ -30,7 +30,6 @@ The command above will output the ID of your Hyperdrive, which you will need to
 
 ```toml
 # required for database drivers to function
```
Comment (thomasgauvin) [inline src/content/docs/hyperdrive/configuration/connect-to-postgres.mdx:32]:
```suggestion
# required for database drivers to function

compatibility_flags = ["nodejs_compat"]
compatibility_date = "2024-09-23"
```

File: src/content/docs/hyperdrive/configuration/connect-to-postgres.mdx [hunk 4]
```diff
@@ -30,7 +30,9 @@ The command above will output the ID of your Hyperdrive, which you will need to
...
-compatibility_flags = [ "nodejs_compat_v2" ]
+
```
Comment (thomasgauvin) [inline src/content/docs/hyperdrive/configuration/connect-to-postgres.mdx]:
```suggestion
```

**Pull request merged**